aif360/algorithms/preprocessing/lfr_helpers/helpers.py

```diff
@@ -38,6 +38,7 @@ def LFR_optim_objective(parameters, x_unprivileged, x_privileged, y_unprivileged
 def get_xhat_y_hat(prototypes, w, x):
     M = softmax(-cdist(x, prototypes), axis=1)
     x_hat = np.matmul(M, prototypes)
+    w = np.clip(w, np.finfo(float).eps, 1.0 - np.finfo(float).eps)
```

Comment:
Constraining the wrights between 0 and 1.

Reply:
with the bnd fix above, I think this becomes unnecessary

Reply:
fixed.